docs/design/partitioning/07-parallel-migrations.md

```diff
@@ -0,0 +1,536 @@
+
+| GET p99 Latency  | GET p99.9 Latency |
+|:-------------:|:-------------:|
+| ![Plot](resources/07/single/master/get/p99.png) | ![Plot](resources/07/single/master/get/p999.png) |
+| GET p99 Min-Max  | GET p99.9 Min-Max |
+| ![Plot](resources/07/single/master/get/p99-min-mean-max.png) | ![Plot](resources/07/single/master/get/p999-min-mean-max.png) |
+
+
+| SET p99 Latency  | SET p99.9 Latency |
+|:-------------:|:-------------:|
+| ![Plot](resources/07/single/master/set/p99.png) | ![Plot](resources/07/single/master/set/p999.png) |
+| SET p99 Min-Max  | SET p99.9 Min-Max |
+| ![Plot](resources/07/single/master/set/p99-min-mean-max.png) | ![Plot](resources/07/single/master/set/p999-min-mean-max.png) |
+
+
+**Heap Usage and GC Pauses**
+
+![Plot](resources/07/single/master/heap.png)
+
+| GC Summary  | |
+|:-------------|-------------:|
+| Full collections |	36 |
+| Mean garbage collection pause (ms) |	89.4 |
+| Mean interval between collections (ms) |	180394 |
+| Proportion of time spent in garbage collection pauses (%) |	3.39 |
+| Proportion of time spent unpaused (%) |	96.61 |
+
+
+---
+
+#### Parallel Migrations - 5x
+
+5x parallelization was used: `hazelcast.partition.max.parallel.migrations=5`.
+
+- *Scale down from 10 to 9 members*: `14000` migrations were completed in **4 minutes and 45 seconds**. 
+Member shutdown in **299 seconds**. 
+
+- *Scale up from 9 to 10 members*: `14000` migrations were completed in **7 minutes and 19 seconds**.
+
+- *Crash single member*: Around `~26150` migrations were completed in **2 minutes and 17 seconds**.
+
+| GET p99 Latency  | GET p99.9 Latency |
+|:-------------:|:-------------:|
+| ![Plot](resources/07/single/parallel-5x/get/p99.png) | ![Plot](resources/07/single/parallel-5x/get/p999.png) |
+| GET p99 Min-Max  | GET p99.9 Min-Max |
+| ![Plot](resources/07/single/parallel-5x/get/p99-min-mean-max.png) | ![Plot](resources/07/single/parallel-5x/get/p999-min-mean-max.png) |
+
+
+| SET p99 Latency  | SET p99.9 Latency |
+|:-------------:|:-------------:|
+| ![Plot](resources/07/single/parallel-5x/set/p99.png) | ![Plot](resources/07/single/parallel-5x/set/p999.png) |
+| SET p99 Min-Max  | SET p99.9 Min-Max |
+| ![Plot](resources/07/single/parallel-5x/set/p99-min-mean-max.png) | ![Plot](resources/07/single/parallel-5x/set/p999-min-mean-max.png) |
+
+
+
+**Heap Usage and GC Pauses**
+
+![Plot](resources/07/single/parallel-5x/heap.png)
+
+| GC Summary  | |
+|:-------------|-------------:|
+| Full collections	| 26
+| Mean garbage collection pause (ms)	| 104
+| Mean interval between collections (ms)	| 56512
+| Proportion of time spent in garbage collection pauses (%)	| 4.27
+| Proportion of time spent unpaused (%)	| 95.73
+
+---
+
+#### Parallel Migrations - 20x
+
+20x parallelization was used: `hazelcast.partition.max.parallel.migrations=20`.
+
+- *Scale down from 10 to 9 members*: `14000` migrations were completed in **3 minutes and 22 seconds**. 
```

Comment:
i can't calculate the improved percentage compared to current master. Congrats for amazing results. From half an hour to 3 minutes. 👏